### PR TITLE
Compilation Fixes for MoveIt on macOS

### PR DESCRIPTION
**Pull request by nkalupahana**
### Description

This pull request fixes various compilation issues on macOS, allowing nearly all of Moveit2 to be compiled on macOS. This build has been verified on CI for macOS 10.15: https://github.com/nkalupahana/ros2-foxy-macos/runs/2815266613?check_suite_focus=true

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/perception/lazy_free_space_updater/CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,9 @@ add_library(${MOVEIT_LIB_NAME} SHARED src/lazy_free_space_updater.cpp)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES COMPILE_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES LINK_FLAGS "${OpenMP_CXX_FLAGS}")
+if(APPLE)
+  target_link_libraries(${MOVEIT_LIB_NAME} OpenMP::OpenMP_CXX)
```
[the remark this answers is not in the document]
Comment (nkalupahana) [reply moveit_ros/perception/lazy_free_space_updater/CMakeLists.txt:8]: Unfortunately, using `ament_target_dependencies(${MOVEIT_LIB_NAME} OpenMP)` leads to a linker error. https://github.com/nkalupahana/ros2-foxy-macos/runs/2841920686?check_suite_focus=true

File: moveit_ros/perception/CMakeLists.txt [hunk 2]
```diff
@@ -21,8 +21,19 @@ if(WITH_OPENGL)
   endif()
   find_package(OpenGL REQUIRED)
   find_package(GLEW REQUIRED)
-  find_package(GLUT REQUIRED)
+  
   set(gl_LIBS ${gl_LIBS} ${OPENGL_LIBRARIES})
+  if(APPLE)
+    find_package(FreeGLUT REQUIRED)
+    set(GLUT_LIBS FreeGLUT::freeglut)
```
[the remark this answers is not in the document]
Comment (nkalupahana) [reply moveit_ros/perception/CMakeLists.txt:28]: Unfortunately, looks like that doesn't work either -- it needs `target_link_dependencies`.
https://github.com/nkalupahana/ros2-foxy-macos/runs/2844121000?check_suite_focus=true

File: moveit_ros/perception/CMakeLists.txt [hunk 3]
```diff
@@ -21,8 +21,19 @@ if(WITH_OPENGL)
...
+    set(GLEW_LIBRARIES
+      ${GLEW_LIBRARIES}
+      GLEW::GLEW
+    )
+  else()
+    find_package(GLUT REQUIRED)
+    set(GLUT_LIBS GLUT::GLUT)
+  endif()
   set(perception_GL_INCLUDE_DIRS "mesh_filter/include" "depth_image_octomap_updater/include")
   set(SYSTEM_GL_INCLUDE_DIRS ${GLEW_INCLUDE_DIR} ${GLUT_INCLUDE_DIR})
```
[the remark this answers is not in the document]
Comment (nkalupahana) [reply moveit_ros/perception/CMakeLists.txt:34]: Alright, did that!

**Comment by nkalupahana** [comment]
@henningkayser fixes are in, please approve the workflows when you get the chance :)

**Comment by nkalupahana** [comment]
@henningkayser can we get this merged in? I've addressed the concerns from your review where possible.

**Comment by tylerjw** [comment]
I approved and started running the CI workflows for this.  Would you mind rebasing this on main so we can get it merged (assuming CI passes)?

**Comment by nkalupahana** [comment]
@tylerjw done! You'll have to re-approve the CI, though.

File: moveit_ros/perception/lazy_free_space_updater/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (tylerjw) [reply moveit_ros/perception/lazy_free_space_updater/CMakeLists.txt:8]: @henningkayser ament_target_dependencies and target_link_libraries do two different things.  ament_target_dependencies adds those dependencies to the list that gets set in the export target, where-as target_link_libraries configures linking.  In this case we might need both, I'm not sure.

[the remark this answers is not in the document]
Comment (henningkayser) [reply moveit_ros/perception/lazy_free_space_updater/CMakeLists.txt:8]: @tylerjw `ament_target_dependencies` calls `target_link_libraries` internally and also configures include directories and other compile definitions. A well configured package should always work with `ament_target_dependencies` if the variables follow conventions. See https://github.com/ament/ament_cmake/blob/master/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake#L21

**Comment by henningkayser** [comment]
@tylerjw please wait for #505 to get merged before merging this, it's always a lot of work to update the sync

File: moveit_ros/perception/lazy_free_space_updater/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (henningkayser) [reply moveit_ros/perception/lazy_free_space_updater/CMakeLists.txt:8]: it's fine if `OpenMP` doesn't work here, though

**Review by henningkayser — APPROVED** [review]

**Pull request merged**